collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -596,20 +602,24 @@ private void refreshView(boolean isGoingUp) {
                         break;
                     }
                     case FormEntryController.EVENT_REPEAT: {
-                        if (!formController.isGroupRelevant()) {
+                        FormEntryCaption fc = formController.getCaptionPrompt();
+                        boolean isFirst = fc.getMultiplicity() == 0;
+                        boolean forPicker = shouldShowRepeatGroupPicker();
+                        boolean isEmpty = !formController.isGroupRelevant();
+                        if (isEmpty &&
+                                // For #4570
```

Comment:
I think if we have a comment here, we'd want to stick it above the `if` so it's not interrupting the logic. Generally we wouldn't bother just linking to an issue as we'll hopefully get that through git history anyway!

Reply:
I actually put it there specifically to emphasise the updated logic, but happy to move it or indeed leave it out. Though I do find comments like this helpful when reading code, as quick hints that there's been an update and why.